javascript/0041-first-missing-positive.js

```diff
@@ -1,47 +1,46 @@
- /**
-   * Time O(n) | Space O(1)
+/**
+ * https://leetcode.com/problems/first-missing-positive/
+ * Time O(n) | Space O(1)
  * @param {number[]} nums
  * @return {number}
  */
 var firstMissingPositive = function(nums) {
     //-1 1 3 4
-    const swap = (a,b) => {
+    const swap = (a, b) => {
```

Comment:
Suggestion: swap in place

```js
const swap = (arr, left, right) => [ arr[left], arr[right] ] = [ arr[right], arr[left] ]
```

Reply:
@aakhtar3, it's done. Wow, cyclicle sort has O(n) time complexity. I didn't know we could do better than n*log(n) for sorting. Just out of curiosity if we can sort it with O(n) with cyclical sort then why don't we use it all the time? I think all programming languages sort with quick sort which is also  n*log(n).